src/main/java/seedu/address/model/person/FriendListPredicate.java

```diff
@@ -8,15 +8,19 @@
  */
 public class FriendListPredicate implements Predicate<Person> {
 
-    private final Set<Friend> friendList;
+    private final Person currentUser;
 
-    public FriendListPredicate(Set<Friend> friendList) {
-        this.friendList = friendList;
+    public FriendListPredicate(Person currentUser) {
+        this.currentUser = currentUser;
     }
 
     @Override
     public boolean test(Person person) {
+        Set<Friend> friendList = currentUser.getFriends();
         for (Friend friend : friendList) {
+            if (currentUser.getName().equals(person.getName())) {
```

Comment:
There's only 3 valid cases, right?

1) Is himself
2) Is friend
3) Is not friend

I think it'd be easier to read if you made each of the 3 cases a branch in an `if-else` statement instead of returning `false` by default